tests/react/async.test.tsx

```diff
@@ -680,37 +658,37 @@ it.skip('a derived atom from a newly created async atom (#351)', async () => {
     )
   }
 
-  render(
-    <>
-      <Suspense fallback="loading">
+  await act(() =>
+    render(
+      <Suspense fallback={<div>loading</div>}>
         <Counter />
-      </Suspense>
-    </>,
+      </Suspense>,
+    ),
   )
 
-  expect(await screen.findByText('loading')).toBeInTheDocument()
-  expect(await screen.findByText('derived: 11, commits: 1')).toBeInTheDocument()
+  expect(screen.getByText('loading')).toBeInTheDocument()
+  await act(() => vi.advanceTimersByTimeAsync(100))
+  expect(screen.getByText('derived: 11, commits: 1')).toBeInTheDocument()
 
-  // The use of fireEvent is required to reproduce the issue
-  fireEvent.click(screen.getByText('button'))
-  expect(await screen.findByText('loading')).toBeInTheDocument()
-  expect(await screen.findByText('derived: 12, commits: 2')).toBeInTheDocument()
+  await act(() => fireEvent.click(screen.getByText('button')))
```

Comment:
Should we wrap fireEvent with act? (I wonder if it reproduces the original issue, hard to test though.)